app/controllers/concerns/foreman/controller/migration_checker.rb

```diff
@@ -0,0 +1,20 @@
+module Foreman::Controller::MigrationChecker
+  extend ActiveSupport::Concern
+
+  included do
+    before_filter :check_pending_migrations
+  end
+
+  private
+
+  def check_pending_migrations
+    if Foreman::Controller::MigrationChecker.needs_migration?
+      raise Foreman::MaintenanceException, _("There are migrations pending in the system.")
```

Comment:
Instructions on how to move forward would be helpful here: `foreman-rake db:migrate`